src/kademlia/traversal_algorithm.cpp

```diff
@@ -593,7 +593,7 @@ void traversal_algorithm::status(dht_lookup& l)
 	l.last_sent = last_sent;
 }
 
-void traversal_algorithm::look_for_nodes(bdecode_node const& r,bool const is_traverse)
+void traversal_algorithm::look_for_nodes(bdecode_node const& r, std::function<void(const node_endpoint&)> f)
```

Comment:
I think you should pass that in as a parameter. The work this function perform does not really affect or relate to the node object. it's parsing a dht message